### DISCUSSION TITLE
Development flow JTAG

### DISCUSSION THREAD

**Discussion by m1dnight** [Mair/esp32-course#199] [General, feedback and testimonials]
Hi all, 

I'm developing in VSCode using an ESP32 Wrover kit. I got everything working as per the videos. I can build, flash, and monitor my device using the buttons in blue  bar at the bottom of vscode. 

However, I often forget to build and flash before I debug the application, and sometimes it just doesn't seem to take at first try. 

Is there any way that I can tell vscode to first build, then flash, and then monitor, before flashing over JTAG? 
Or are there other ways to make the process a bit smoother.

Thanks!

**Comment by Mair** [comment]
You can write a script using the IDF commands to do all the above in sequence. PowerShell or batch for windows or bash for Linux and Mac work well
